src/SpeedDial/SpeedDial.js

```diff
@@ -0,0 +1,198 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { findDOMNode } from 'react-dom';
+import keycode from 'keycode';
+import { withStyles } from '../styles';
+import Button from '../Button';
+import Icon from '../Icon';
+import AddIcon from '../svg-icons/Add';
+
+const styles = theme => ({
+  root: {
+    width: 56,
+    textAlign: 'center',
+    zIndex: theme.zIndex.speedDial,
+    display: 'flex',
+    flexDirection: 'column-reverse', // Place the Actions above the FAB.
+  },
+  fab: {
+    transition: theme.transitions.create('transform', {
+      duration: theme.transitions.duration.shortest,
+    }),
+  },
+  hiddenFab: {
+    transform: 'scale3d(0, 0, 1)',
+  },
+  fabIcon: {
+    transition: theme.transitions.create('transform', {
+      duration: theme.transitions.duration.shorter,
+      easing: theme.transitions.easing.easeInOut,
+    }),
+  },
+  openFabIcon: {
+    transform: 'rotate(135deg)',
+  },
+  actions: {
+    display: 'flex',
+    flexDirection: 'column-reverse', // Display the first action at the bottom.
+    marginBottom: 16,
+  },
+  closedActions: {
+    transition: 'top 0s linear 0.2s',
+  },
+});
+
+class SpeedDial extends React.Component {
+  state = {
+    nextKey: 'up',
+    prevKey: 'down',
+  };
+
+  handleKeyDown = (event: SyntheticUIEvent<>) => {
```

Comment:
Maybe we can extract and share the same logic than with the MenuList. Right now, something is off. It doesn't loop correctly. I can get into a situation where the up and down arrow keys have a reverse effect.

Reply:
That is by design. See: https://github.com/mui-org/material-ui/pull/9539#issuecomment-353171073